### ISSUE TITLE
More XCM e2e tests

### ISSUE THREAD

**Issue by xlc** [paritytech/polkadot#7070]
XCM v2 messages with limited weight isn't working for Polkadot due to https://github.com/paritytech/polkadot/issues/6785 and this is a regression but we only found it out in our pre runtime upgrade manual testing.

This means we need more tests.

**Comment by ggwpez** [comment]
Do you override the Polkadot runtime WASM before running these tests?  
Then we could do it on every MR with the patched runtime.

**Comment by bkchr** [comment]
CC @s3krit 

(@s3krit could you maybe create some kind of integration testing project to keep track of these issues?)

**Comment by xlc** [comment]
> Do you override the Polkadot runtime WASM before running these tests?

Yes. We setup new testnet with Chopsticks with new Polkadot runtime and new Acala runtime to test compatibility.

We also have this as automated e2e test and plan to gradually add more test cases.

https://github.com/AcalaNetwork/e2e-tests/blob/062c65122b63af7f7a322680df263ce8d68be436/tests/xcm-transfer/polkadot-relay.test.ts#L32

**Comment by bkchr** [comment]
Nice @xlc. @s3krit maybe a good way forward would also be to start a topic in the forum to collect all the different integration testing frameworks that already exist out there!
